modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/SiteResourceImpl.java

```diff
@@ -643,21 +717,144 @@ else if (membershipType.equals(Site.MembershipType.PRIVATE)) {
 	}
 
 	private String _getTypeSettings(
-			Map<String, String> typeSettings,
-			UnicodeProperties oldUnicodeProperties)
-		throws Exception {
+		Site site, UnicodeProperties oldUnicodeProperties) {
 
-		if (typeSettings == null) {
-			return null;
+		UnicodeProperties unicodeProperties = new UnicodeProperties();
+
+		AnalyticsConfiguration analyticsConfiguration =
+			site.getAnalyticsConfiguration();
+
+		if (analyticsConfiguration != null) {
+			GoogleAnalyticsConfiguration googleAnalyticsConfiguration =
+				analyticsConfiguration.getGoogleAnalyticsConfiguration();
+
+			if (googleAnalyticsConfiguration != null) {
+				unicodeProperties.put(
+					"googleAnalytics4CustomConfiguration",
+					googleAnalyticsConfiguration.getGoogleAnalytics4Config());
+				unicodeProperties.put(
+					"googleAnalytics4Id",
+					googleAnalyticsConfiguration.getGoogleAnalytics4Id());
+				unicodeProperties.put(
+					"googleAnalyticsCustomConfiguration",
+					googleAnalyticsConfiguration.getGoogleAnalyticsConfig());
+				unicodeProperties.put(
+					"googleAnalyticsCreateCustomConfiguration",
+					googleAnalyticsConfiguration.
+						getGoogleAnalyticsCreateMethodConfig());
+				unicodeProperties.put(
+					"googleAnalyticsId",
+					googleAnalyticsConfiguration.getGoogleAnalyticsId());
+			}
+
+			unicodeProperties.put(
+				"analytics_matomo",
+				analyticsConfiguration.getMatomoAnalyticsScript());
 		}
 
-		UnicodeProperties unicodeProperties = UnicodePropertiesBuilder.putAll(
-			typeSettings
-		).build();
+		unicodeProperties.put(
+			"assetAutoTaggingEnabled",
+			String.valueOf(site.getAssetAutoTaggingEnabled()));
+		unicodeProperties.put(
+			"locales",
+			StringUtil.merge(site.getAvailableLanguages(), StringPool.COMMA));
+		unicodeProperties.put(
+			"contentSharingWithChildrenEnabled",
+			String.valueOf(site.getContentSharingWithChildrenEnabled()));
```

Comment:
This value uses integer constants and not a boolean - https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/sites/kernel/util/Sites.java#L27-L37

From what I can tell though, for a Group it only uses the values `-1`, `0`, and `3`. Perhaps we can simply transform `false` to `0` and `true` to `3`?